# 🪵🍁🍂🍄🍗🦃🏈 Big Thanksgiving Dinner Analysis 🏈🦃🍗🍄🍂🍁🪵
**Name:** Charlize Gutierrez-Lopez        <br>
**Class:** 12pm

In [1]:
# Libraries

import numpy as np
import pandas as pd


In [2]:
# Synthetic Thanksgiving Data
np.random.seed(2310)    # set seed to 2310
n =   50  # set n to 50

# creating the variables first (Age, Dinner, Dessert, Duration)
Age = (np.random.normal(37, 18, n).round(0).clip(5, 90))
Num_Dinner_Servings = np.round(4 - 0.0009 * (Age - 40)**2 + np.random.normal(0, 0.3, n)).clip(1, 5)
Num_Dessert_Servings = np.round(np.clip(2.5 - (Age / 50) + np.random.normal(0, 0.5, n), 0, 3))
Dinner_Duration_Hours =  np.round(1.5 + (Age / 40) + np.random.normal(0, 0.5, n), 1).clip(1, 5)

# creating boolean for whether or not they watched tv (older and younger people more likely to watch)
prob_parade = np.clip(    0.3 + 0.4 * np.exp(-((Age - 15)/20)**2) + 0.4 * np.exp(-((Age - 70)/20)**2),0, 1)
Watched_Parade = np.random.binomial(1, prob_parade)

# creating the data frame using the variables we just made
Thanksgiving = pd.DataFrame({
    'Ages' : Age,
    'Dinner_Servings' : Num_Dinner_Servings,
    'Dessert_Servings' : Num_Dessert_Servings,
    'Duration' : Dinner_Duration_Hours,
    'Macy_Parade' : Watched_Parade
})

Thanksgiving.head()


Ages  Dinner_Servings  Dessert_Servings  Duration  Macy_Parade
0  33.0              4.0               1.0       2.0            1
1  36.0              4.0               2.0       2.2            0
2  37.0              4.0               2.0       3.0            0
3  31.0              4.0               2.0       2.2            1
4  34.0              5.0               2.0       2.0            1

In [3]:
# Use the `describe` function to get the basic summary statistics

Thanksgiving.describe()

Ages  Dinner_Servings  Dessert_Servings  Duration  Macy_Parade
count  50.000000        50.000000         50.000000  50.00000    50.000000
mean   39.520000         3.920000          1.500000   2.50400     0.680000
std    14.816207         0.395897          0.646813   0.56531     0.471212
min     9.000000         3.000000          0.000000   1.00000     0.000000
25%    31.250000         4.000000          1.000000   2.20000     0.000000
50%    37.500000         4.000000          1.500000   2.55000     1.000000
75%    50.750000         4.000000          2.000000   2.90000     1.000000
max    72.000000         5.000000          3.000000   3.50000     1.000000

In [5]:
print(2.50400+2*.56531)


3.63462


## Instructions:
Use this Thanksgiving dataset to explore patterns in the family dining habits.
You’ll need to _**interpret**_ the data using summary statistics, **not just report them**.

You will being using **only the describe function** to answer your questions

### Questions:

1. Looking at the minimum and maximum values of Age, what do they suggest about who attended this Thanksgiving dinner? <br>

   
**Answer:** Family members of all ages appear to have come to the Thanksgiving dinner, with the youngest being 9 and the oldest being 72. <br>

3. Looking at the mean and standard deviation of Dinner_Duration, what do they show about how consistent the mealtime was across family members? <br>

   
**Answer:** The mealtime was relatively consistent, usually around 2 to 3 hours. <br>

5. If someone stayed for 4.8 hours, would you consider them an outlier? _Use the mean and standard deviation to support your answer (Hint: $mean + 2*std$ OR $IQR=Q3−Q1$ and $Upper Bound=Q3+1.5×IQR$)._ <br>

   
**Answer:** I would consider that an outlier. 2.50400+2*.56531 is equal to 3.63462, so anything above this is an outlier.


7. Compare the mean and median of Dinner_Servings. What could a large difference between them indicate about family member behavior? _(Hint: 50% is half way in the data set)_ <br>

   
**Answer:** This large difference could indicate that there are many family members who had dinner servings that could be considered outliers. <br>

9. Looking at the max value of Dessert_Servings, what type of guest do you imagine that might be and what could this tell us about dessert consumption overall?  <br>

    
**Answer:** Max value of dessert servings is 3. I imagine its the younger kids who had multiple servings of dessert. This could also tell us that the family members have a sweet tooth.  <br>

11. Suppose the standard deviation of Dinner_Serving is small. What does that suggest about how much everyone ate?  <br>
    
**Answer:** If the standard deviation of Dinner_Serving is small, that means there are not many outliers and everyone ate around the same. <br>

12. Which statistic (min, max, mean, std) best helps you find the “light eaters”? Explain how you would identify them and if the minimum makes sense.<br>


**Answer:** Min of dinner servings would help you find the light eaters. I would identify them by the least amount of servings, this would make sense since they eat the least. <br>

13. The range (max − min) of Dinner_Duration is quite large. What could that imply about how the dinner was organized?  <br>

    
**Answer:**   There was high variation in the times of the dinners, so there maybe was no plan on how the dinner would go. <br>

15. You suspect that older guests tend to eat fewer desserts. Could you use only the `describe` function to explore whether that idea might be true before doing a full analysis or would we have to include something else?  <br>
    
**Answer:** I would include a correlation matrix for ages and desserts, the describe function does not really explore that correlation. <br>

17. What does the mean tell you about how common it was for people to watch the Macy’s Day Parade?  <br>
    
**Answer:** The mean is .68, with a max of 1, and a min of 0. The mean tells me that more than half of people watched the Macy's parade so it is common.<br>

19. What does the standard deviation suggest about the variation in watching behavior? Was parade-watching behavior fairly consistent across people, or was there a lot of difference between individuals?  <br>

    
**Answer:**  There was not much standard deviation in watching behavior, was fairly consistent across people.<br>

21. In one sentence, summarize what kind of Thanksgiving this dataset represents. Focus on what the numbers imply rather than just stating the numbers.  <br>

    
**Answer:**  This is a chaotic Thanksgiving with a lot of different family and a lot of food, usually followed by a Macy's watchparty. <br>

23. Suggest some visuals that would be ideal for exploring this data and state their purpose. <br>
    
**Answer:** Some visuals that would be ideal for exploring is a bar plot that compares ages and dessert servings to see frequency and trends of the sweet tooth someone has and their age, and histograms of dinner servings vs dessert servings to see if the people who ate more food ate less (or more) dessert. <br>
